### ISSUE TITLE
[Error] Could not find a connection between 'iiwa_link_0' and 'tracking_marker'

### ISSUE THREAD

**Issue by lyh458** [marcoesposito1988/easy_handeye_demo#7]
Hi,
although with issue #6 , I calibrated successfully. But when I run the command ``roslaunch easy_handeye_demo check_calibration.launch eye_on_hand:=true`` to check the calibration, several warnings and errors occurred, here is the log:
```
src/easy_handeye/easy_handeye/src/easy_handeye/handeye_calibration.py:177: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.

![](https://i.loli.net/2020/12/01/SAMcwOUVsdmbHKj.png)
![](https://i.loli.net/2020/12/01/mbxnTD4MwdBCkf1.png)

... logging to /home/lyh/.ros/log/6d865ffa-3382-11eb-ae4b-e09d31e91a88/roslaunch-birl-lyh-pc-12342.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

xacro: in-order processing became default in ROS Melodic. You can drop the option.
started roslaunch server http://birl-lyh-pc:41165/

SUMMARY
========

PARAMETERS
 * /demo_handeyecalibration_eye_on_hand/calibration_evaluator/eye_on_hand: True
 * /demo_handeyecalibration_eye_on_hand/calibration_evaluator/robot_base_frame: iiwa_link_0
 * /demo_handeyecalibration_eye_on_hand/calibration_evaluator/robot_effector_frame: iiwa_link_ee
 * /demo_handeyecalibration_eye_on_hand/calibration_evaluator/tracking_base_frame: tracking_origin
 * /demo_handeyecalibration_eye_on_hand/calibration_evaluator/tracking_marker_frame: tracking_marker
 * /demo_handeyecalibration_eye_on_hand/eye_on_hand: True
 * /demo_handeyecalibration_eye_on_hand/inverse: False
 * /demo_handeyecalibration_eye_on_hand/robot_effector_frame: iiwa_link_ee
 * /demo_handeyecalibration_eye_on_hand/tracking_base_frame: tracking_origin
 * /iiwa/joint_state_publisher/source_list: ['/move_group/fak...
 * /iiwa/joint_state_publisher/use_gui: False
 * /iiwa/move_group/allow_trajectory_execution: True
 * /iiwa/move_group/capabilities: move_group/MoveGr...
 * /iiwa/move_group/controller_list: [{'joints': ['iiw...
 * /iiwa/move_group/jiggle_fraction: 0.05
 * /iiwa/move_group/manipulator/planner_configs: ['SBLkConfigDefau...
 * /iiwa/move_group/max_range: 5.0
 * /iiwa/move_group/max_safe_path_cost: 1
 * /iiwa/move_group/moveit_controller_manager: moveit_fake_contr...
 * /iiwa/move_group/moveit_manage_controllers: True
 * /iiwa/move_group/octomap_resolution: 0.025
 * /iiwa/move_group/planner_configs/BKPIECEkConfigDefault/border_fraction: 0.9
 * /iiwa/move_group/planner_configs/BKPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /iiwa/move_group/planner_configs/BKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /iiwa/move_group/planner_configs/BKPIECEkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/BKPIECEkConfigDefault/type: geometric::BKPIECE
 * /iiwa/move_group/planner_configs/ESTkConfigDefault/goal_bias: 0.05
 * /iiwa/move_group/planner_configs/ESTkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/ESTkConfigDefault/type: geometric::EST
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/border_fraction: 0.9
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/goal_bias: 0.05
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/KPIECEkConfigDefault/type: geometric::KPIECE
 * /iiwa/move_group/planner_configs/LBKPIECEkConfigDefault/border_fraction: 0.9
 * /iiwa/move_group/planner_configs/LBKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /iiwa/move_group/planner_configs/LBKPIECEkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/LBKPIECEkConfigDefault/type: geometric::LBKPIECE
 * /iiwa/move_group/planner_configs/PRMkConfigDefault/max_nearest_neighbors: 10
 * /iiwa/move_group/planner_configs/PRMkConfigDefault/type: geometric::PRM
 * /iiwa/move_group/planner_configs/PRMstarkConfigDefault/type: geometric::PRMstar
 * /iiwa/move_group/planner_configs/RRTConnectkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/RRTConnectkConfigDefault/type: geometric::RRTCon...
 * /iiwa/move_group/planner_configs/RRTkConfigDefault/goal_bias: 0.05
 * /iiwa/move_group/planner_configs/RRTkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/RRTkConfigDefault/type: geometric::RRT
 * /iiwa/move_group/planner_configs/RRTstarkConfigDefault/delay_collision_checking: 1
 * /iiwa/move_group/planner_configs/RRTstarkConfigDefault/goal_bias: 0.05
 * /iiwa/move_group/planner_configs/RRTstarkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/RRTstarkConfigDefault/type: geometric::RRTstar
 * /iiwa/move_group/planner_configs/SBLkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/SBLkConfigDefault/type: geometric::SBL
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/frountierNodeRatio: 0.1
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/frountier_threshold: 0.0
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/goal_bias: 0.05
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/init_temperature: 10e-6
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/k_constant: 0.0
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/max_states_failed: 10
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/min_temperature: 10e-10
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/range: 0.0
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/temp_change_factor: 2.0
 * /iiwa/move_group/planner_configs/TRRTkConfigDefault/type: geometric::TRRT
 * /iiwa/move_group/planning_plugin: ompl_interface/OM...
 * /iiwa/move_group/planning_scene_monitor/publish_geometry_updates: True
 * /iiwa/move_group/planning_scene_monitor/publish_planning_scene: True
 * /iiwa/move_group/planning_scene_monitor/publish_state_updates: True
 * /iiwa/move_group/planning_scene_monitor/publish_transforms_updates: True
 * /iiwa/move_group/request_adapters: default_planner_r...
 * /iiwa/move_group/start_state_max_bounds_error: 0.1
 * /iiwa/move_group/trajectory_execution/allowed_execution_duration_scaling: 1.2
 * /iiwa/move_group/trajectory_execution/allowed_goal_duration_margin: 0.5
 * /iiwa/robot_description: <?xml version="1....
 * /iiwa/robot_description_kinematics/manipulator/kinematics_solver: kdl_kinematics_pl...
 * /iiwa/robot_description_kinematics/manipulator/kinematics_solver_attempts: 3
 * /iiwa/robot_description_kinematics/manipulator/kinematics_solver_search_resolution: 0.005
 * /iiwa/robot_description_kinematics/manipulator/kinematics_solver_timeout: 0.005
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_1/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_1/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_1/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_1/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_2/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_2/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_2/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_2/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_3/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_3/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_3/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_3/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_4/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_4/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_4/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_4/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_5/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_5/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_5/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_5/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_6/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_6/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_6/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_6/max_velocity: 10
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_7/has_acceleration_limits: False
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_7/has_velocity_limits: True
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_7/max_acceleration: 0
 * /iiwa/robot_description_planning/joint_limits/iiwa_joint_7/max_velocity: 10
 * /iiwa/robot_description_semantic: <?xml version="1....
 * /rosdistro: melodic
 * /rosversion: 1.14.10
 * /tracking_simulator/arbitrary_marker_placement_transformation: 1 0 0.5 0 0 0 1
 * /tracking_simulator/eye_on_hand: True
 * /tracking_simulator/frequency: 10
 * /tracking_simulator/ground_truth_calibration_transformation: 0.12 0.21 0.137 0...
 * /tracking_simulator/is_calibration: False
 * /tracking_simulator/robot_base_frame: iiwa_link_0
 * /tracking_simulator/robot_effector_frame: iiwa_link_ee
 * /tracking_simulator/rotation_noise_stdev: 1e-05
 * /tracking_simulator/tracking_base_frame: tracking_origin
 * /tracking_simulator/tracking_marker_frame: tracking_marker
 * /tracking_simulator/translation_noise_stdev: 0.0001

NODES
  /
    tracking_simulator (easy_handeye_demo/tracking_simulator.py)
  /demo_handeyecalibration_eye_on_hand/
    calibration_evaluator (rqt_easy_handeye/rqt_calibration_evaluator)
    handeye_publisher_birl_lyh_pc_12342_4844057588746556651 (easy_handeye/publish.py)
  /iiwa/
    joint_state_publisher (joint_state_publisher/joint_state_publisher)
    move_group (moveit_ros_move_group/move_group)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    rviz_birl_lyh_pc_12342_966738794670550594 (rviz/rviz)

auto-starting new master
process[master]: started with pid [12361]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 6d865ffa-3382-11eb-ae4b-e09d31e91a88
process[rosout-1]: started with pid [12372]
started core service [/rosout]
process[iiwa/joint_state_publisher-2]: started with pid [12379]
process[iiwa/robot_state_publisher-3]: started with pid [12380]
process[iiwa/move_group-4]: started with pid [12381]
process[tracking_simulator-5]: started with pid [12382]
process[demo_handeyecalibration_eye_on_hand/handeye_publisher_birl_lyh_pc_12342_4844057588746556651-6]: started with pid [12389]
process[demo_handeyecalibration_eye_on_hand/calibration_evaluator-7]: started with pid [12390]
process[iiwa/rviz_birl_lyh_pc_12342_966738794670550594-8]: started with pid [12391]
[ INFO] [1606792086.894215752]: Loading robot model 'iiwa7'...
[ INFO] [1606792086.894859717]: No root/virtual joint specified in SRDF. Assuming fixed joint
[ INFO] [1606792087.139674561]: rviz version 1.13.14
[ INFO] [1606792087.139719051]: compiled against Qt version 5.9.5
[ INFO] [1606792087.139730435]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1606792087.148384207]: Forcing OpenGl version 0.
[ WARN] [1606792087.596818824]: Kinematics solver doesn't support #attempts anymore, but only a timeout.
Please remove the parameter '/iiwa/robot_description_kinematics/manipulator/kinematics_solver_attempts' from your configuration.
[ INFO] [1606792087.849562103]: Stereo is NOT SUPPORTED
[ INFO] [1606792087.849640120]: OpenGl version: 3 (GLSL 1.3).
[ INFO] [1606792087.855133056]: Publishing maintained planning scene on 'monitored_planning_scene'
[ INFO] [1606792087.857390610]: MoveGroup debug mode is OFF
Starting planning scene monitors...
[ INFO] [1606792087.857427233]: Starting planning scene monitor
[ INFO] [1606792087.859383575]: Listening to '/iiwa/planning_scene'
[ INFO] [1606792087.859422083]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[ INFO] [1606792087.861343662]: Listening to '/iiwa/collision_object'
[ INFO] [1606792087.862882591]: Listening to '/iiwa/planning_scene_world' for planning scene world geometry
[ERROR] [1606792087.863577645]: Failed to find 3D sensor plugin parameters for octomap generation
[ INFO] [1606792088.003109668]: Listening to '/iiwa/attached_collision_object' for attached collision objects
Planning scene monitors started.
[ INFO] [1606792088.045633181]: Initializing OMPL interface using ROS parameters
[ INFO] [1606792088.059379915]: Using planning interface 'OMPL'
[ INFO] [1606792088.065414270]: Param 'default_workspace_bounds' was not set. Using default value: 10
[ INFO] [1606792088.065837988]: Param 'start_state_max_bounds_error' was set to 0.1
[ INFO] [1606792088.066142678]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1606792088.066451572]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1606792088.066751799]: Param 'jiggle_fraction' was set to 0.05
[ INFO] [1606792088.067026832]: Param 'max_sampling_attempts' was not set. Using default value: 100
[ INFO] [1606792088.067109761]: Using planning request adapter 'Add Time Parameterization'
[ INFO] [1606792088.067163651]: Using planning request adapter 'Fix Workspace Bounds'
[ INFO] [1606792088.067184612]: Using planning request adapter 'Fix Start State Bounds'
[ INFO] [1606792088.067199677]: Using planning request adapter 'Fix Start State In Collision'
[ INFO] [1606792088.067211463]: Using planning request adapter 'Fix Start State Path Constraints'
[ INFO] [1606792088.077364327]: Fake controller 'fake_manipulator_controller' with joints [ iiwa_joint_1 iiwa_joint_2 iiwa_joint_3 iiwa_joint_4 iiwa_joint_5 iiwa_joint_6 iiwa_joint_7 ]
[ INFO] [1606792088.078192138]: Returned 1 controllers in list
[ INFO] [1606792088.090994235]: Trajectory execution is managing controllers
Loading 'move_group/ApplyPlanningSceneService'...
Loading 'move_group/ClearOctomapService'...
Loading 'move_group/MoveGroupCartesianPathService'...
Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
Loading 'move_group/MoveGroupGetPlanningSceneService'...
Loading 'move_group/MoveGroupKinematicsService'...
Loading 'move_group/MoveGroupMoveAction'...
Loading 'move_group/MoveGroupPickPlaceAction'...
Loading 'move_group/MoveGroupPlanService'...
Loading 'move_group/MoveGroupQueryPlannersService'...
Loading 'move_group/MoveGroupStateValidationService'...
[ INFO] [1606792088.136141138]: 

********************************************************
* MoveGroup using: 
*     - ApplyPlanningSceneService
*     - ClearOctomapService
*     - CartesianPathService
*     - ExecuteTrajectoryAction
*     - GetPlanningSceneService
*     - KinematicsService
*     - MoveAction
*     - PickPlaceAction
*     - MotionPlanService
*     - QueryPlannersService
*     - StateValidationService
********************************************************

[ INFO] [1606792088.136208004]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[ INFO] [1606792088.136235024]: MoveGroup context initialization complete

You can start planning now!

/mnt/WinDisk/Users/liyih/OneDrive - mail2.gdut.edu.cn/Ph.D/Codes/calibration_ws/src/easy_handeye/easy_handeye/src/easy_handeye/handeye_calibration.py:177: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  return HandeyeCalibration.from_dict(yaml.load(in_yaml))
Traceback (most recent call last):
  File "/home/lyh/Codes/calibration_ws/src/easy_handeye/easy_handeye/scripts/publish.py", line 15, in <module>
    calib = HandeyeCalibration.from_file(rospy.get_namespace())
  File "/mnt/WinDisk/Users/liyih/OneDrive - mail2.gdut.edu.cn/Ph.D/Codes/calibration_ws/src/easy_handeye/easy_handeye/src/easy_handeye/handeye_calibration.py", line 254, in from_file
    return HandeyeCalibration.from_yaml(calib_file.read())
  File "/mnt/WinDisk/Users/liyih/OneDrive - mail2.gdut.edu.cn/Ph.D/Codes/calibration_ws/src/easy_handeye/easy_handeye/src/easy_handeye/handeye_calibration.py", line 177, in from_yaml
    return HandeyeCalibration.from_dict(yaml.load(in_yaml))
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 45, in get_single_data
    return self.construct_document(node)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 54, in construct_document
    for dummy in generator:
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 404, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 214, in construct_mapping
    return BaseConstructor.construct_mapping(self, node, deep=deep)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 139, in construct_mapping
    value = self.construct_object(value_node, deep=deep)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 96, in construct_object
    data = constructor(self, tag_suffix, node)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 605, in construct_python_object_apply
    args = self.construct_sequence(node, deep=True)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 124, in construct_sequence
    for child in node.value]
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 96, in construct_object
    data = constructor(self, tag_suffix, node)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 617, in construct_python_object_apply
    instance = self.make_python_instance(suffix, node, args, kwds, newobj)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 554, in make_python_instance
    cls = self.find_python_name(suffix, node.start_mark)
  File "/home/lyh/anaconda3/envs/ros/lib/python2.7/site-packages/yaml/constructor.py", line 522, in find_python_name
    "module %r is not imported" % module_name.encode('utf-8'), mark)
yaml.constructor.ConstructorError: while constructing a Python object
module 'numpy' is not imported
  in "<string>", line 12, column 5:
      - &id001 !!python/object/apply:num ... 
        ^
[INFO] [1606792088.490119]: Waiting for tf tree to be connected...
[demo_handeyecalibration_eye_on_hand/handeye_publisher_birl_lyh_pc_12342_4844057588746556651-6] process has died [pid 12389, exit code 1, cmd /home/lyh/Codes/calibration_ws/src/easy_handeye/easy_handeye/scripts/publish.py __name:=handeye_publisher_birl_lyh_pc_12342_4844057588746556651 __log:=/home/lyh/.ros/log/6d865ffa-3382-11eb-ae4b-e09d31e91a88/demo_handeyecalibration_eye_on_hand-handeye_publisher_birl_lyh_pc_12342_4844057588746556651-6.log].
log file: /home/lyh/.ros/log/6d865ffa-3382-11eb-ae4b-e09d31e91a88/demo_handeyecalibration_eye_on_hand-handeye_publisher_birl_lyh_pc_12342_4844057588746556651-6*.log
arguments:  Namespace(quiet=False)
unknowns:  []
[INFO] [1606792090.393554]: Could not sample pose!
[ERROR] [1606792090.397756]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792090.819201]: Could not sample pose!
[ERROR] [1606792090.822362]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[ INFO] [1606792091.220073051]: Loading robot model 'iiwa7'...
[ INFO] [1606792091.220123599]: No root/virtual joint specified in SRDF. Assuming fixed joint
[INFO] [1606792091.293944]: Could not sample pose!
[ERROR] [1606792091.295928]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[ WARN] [1606792091.298642899]: Kinematics solver doesn't support #attempts anymore, but only a timeout.
Please remove the parameter '/iiwa/robot_description_kinematics/manipulator/kinematics_solver_attempts' from your configuration.
[ INFO] [1606792091.588370770]: Starting planning scene monitor
[ INFO] [1606792091.590579578]: Listening to '/iiwa/move_group/monitored_planning_scene'
[INFO] [1606792091.818328]: Could not sample pose!
[ERROR] [1606792091.819858]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[ WARN] [1606792091.947085443]: Unable to transform object from frame 'tracking_marker' to planning frame 'world' (Could not find a connection between 'world' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.)
[ WARN] [1606792091.947157381]: Unable to transform object from frame 'tracking_origin' to planning frame 'world' (Could not find a connection between 'world' and 'tracking_origin' because they are not part of the same tree.Tf has two or more unconnected trees.)
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
[ INFO] [1606792091.947996966]: Constructing new MoveGroup connection for group 'manipulator' in namespace ''
[INFO] [1606792092.293753]: Could not sample pose!
[ERROR] [1606792092.298297]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792092.792801]: Could not sample pose!
[ERROR] [1606792092.797102]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[ INFO] [1606792093.082830056]: Ready to take commands for planning group manipulator.
[ INFO] [1606792093.083028640]: Looking around: no
[ INFO] [1606792093.083177477]: Replanning: no
[INFO] [1606792093.292941]: Could not sample pose!
[ERROR] [1606792093.297554]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792093.792697]: Could not sample pose!
[ERROR] [1606792093.797237]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792094.291860]: Could not sample pose!
[ERROR] [1606792094.296719]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792094.792540]: Could not sample pose!
[ERROR] [1606792094.797022]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792095.292031]: Could not sample pose!
[ERROR] [1606792095.295982]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792095.792852]: Could not sample pose!
[ERROR] [1606792095.801210]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792096.292262]: Could not sample pose!
[ERROR] [1606792096.296679]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792096.792012]: Could not sample pose!
[ERROR] [1606792096.797488]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792097.292693]: Could not sample pose!
[ERROR] [1606792097.297417]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792097.792676]: Could not sample pose!
[ERROR] [1606792097.797185]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792098.291748]: Could not sample pose!
[ERROR] [1606792098.296458]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792098.791924]: Could not sample pose!
[ERROR] [1606792098.795996]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792099.291900]: Could not sample pose!
[ERROR] [1606792099.296448]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792099.792528]: Could not sample pose!
[ERROR] [1606792099.796868]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792100.292292]: Could not sample pose!
[ERROR] [1606792100.296918]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792100.792828]: Could not sample pose!
[ERROR] [1606792100.801637]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792101.292619]: Could not sample pose!
[ERROR] [1606792101.297937]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
[INFO] [1606792101.791770]: Could not sample pose!
[ERROR] [1606792101.796257]: Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees.
^C[iiwa/rviz_birl_lyh_pc_12342_966738794670550594-8] killing on exit
[demo_handeyecalibration_eye_on_hand/calibration_evaluator-7] killing on exit
[tracking_simulator-5] killing on exit
[iiwa/move_group-4] killing on exit
[iiwa/robot_state_publisher-3] killing on exit
[iiwa/joint_state_publisher-2] killing on exit
[INFO] [1606792102.193559]: Could not sample pose!
[ERROR] [1606792102.194987]: Lookup would require extrapolation into the future.  Requested time 1606792101.993136883 but the latest data is at time 1606792101.883462906, when looking up transform from frame [iiwa_link_ee] to frame [iiwa_link_0]
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done
```
They mainly are:

- error 1
```
[ERROR] [1606792087.863577645]: Failed to find 3D sensor plugin parameters for octomap generation
```

- warning 1
```
YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
```
which can be solved by changing relative function as ``return HandeyeCalibration.from_dict(yaml.load(in_yaml, Loader=yaml.FullLoader))``, see [PyYAML yaml.load(input) Deprecation](https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation), I will create a pull request to [easy_handeye](https://github.com/IFL-CAMP/easy_handeye) later.

- error 2
``1
"module %r is not imported" % module_name.encode('utf-8'), mark) yaml.constructor.ConstructorError: while constructing a Python object module 'numpy' is not imported``

- error 3
```
[demo_handeyecalibration_eye_on_hand/handeye_publisher_birl_lyh_pc_12342_4844057588746556651-6] process has died
```

- error 4
```
Could not find a connection between 'iiwa_link_0' and 'tracking_marker' because they are not part of the same tree.Tf has two or more unconnected trees
```
Thanks.

**Comment by lyh458** [comment]
And the two showed windows after run the command:
![](https://i.loli.net/2020/12/01/SAMcwOUVsdmbHKj.png)
![](https://i.loli.net/2020/12/01/mbxnTD4MwdBCkf1.png)

**Comment by marcoesposito1988** [comment]
Thanks for reporting, this bug was introduced in 0.4.0. I released a new version of easy_handeye (0.4.1) that fixes this issue. You will have to pull from that repository and calibrate again.

Sorry about that!

**Comment by lyh458** [comment]
> Thanks for reporting, this bug was introduced in 0.4.0. I released a new version of easy_handeye (0.4.1) that fixes this issue. You will have to pull from that repository and calibrate again.
> 
> Sorry about that!

Thanks! It works after merging the update.
